src/Checkpoints.jl

```diff
@@ -24,9 +25,27 @@ __init__() = Memento.register(LOGGER)
 include("handler.jl")
 
 const CHECKPOINTS = Dict{String, Union{Nothing, Handler}}()
+@contextvar TAGS = OrderedDict{Symbol, Any}()
 
 include("session.jl")
 
+"""
+    with_tags(f::Function, tags::Pair...)
```

Comment:
how do we feel about this name?

Reply:
I am ok with it but happy to change if a better one comes up